### ISSUE TITLE
assertThat(...).as(...).isNotNull() is not recognized

### ISSUE THREAD

**Issue by vladykin** [uber/NullAway#877]
NullAway recognizes the `assertThat(...).isNotNull()` (https://github.com/uber/NullAway/issues/301), but fails to recognize if the description is added. Please refer to the example below. NullAway version is 0.10.18.

```java
package com.example;

import static org.assertj.core.api.Assertions.assertThat;

import jakarta.annotation.Nullable;
import org.junit.jupiter.api.Test;

public class NullAwayTest {

  @Test
  void assertNotNull() {
    String s = getString();
    assertThat(s).isNotNull();
    assertThat(s.length()).isEqualTo(0); // NullAway doesn't complain, this is correct
  }

  @Test
  void assertNotNullAs() {
    String s = getString();
    assertThat(s).as("text").isNotNull();
    assertThat(s.length()).isEqualTo(0); // NullAway complains, this is wrong
  }

  @Test
  void assertNotNullDescribedAs() {
    String s = getString();
    assertThat(s).describedAs("text").isNotNull();
    assertThat(s.length()).isEqualTo(0); // NullAway complains, this is wrong
  }

  static @Nullable String getString() {
    return null;
  }
}
```

**Comment by msridhar** [comment]
Thanks for the report.  The relevant code in NullAway is here:

https://github.com/uber/NullAway/blob/5fbee1ff8682762cd2f291070491fe3e476787cf/nullaway/src/main/java/com/uber/nullaway/handlers/AssertionHandler.java#L59-L75

The logic is explicitly looking for an AST corresponding to `assertThat(A).isNotNull()`.  Instead, we'd have to generalize this logic to recurse when appropriate to look for a more deeply-nested `assertThat` call.

@vladykin is it the case that any chain of method calls starting with `assertThat(s)` and ending in `isNotNull()` should make `s` `@NonNull`?  Or is it only for specific intermediate method calls like `as()` or `describedAs()`?

We would welcome a PR for this one.

**Comment by vladykin** [comment]
> is it the case that any chain of method calls starting with `assertThat(s)` and ending in `isNotNull()` should make s `@NonNull`? Or is it only for specific intermediate method calls like `as()` or `describedAs()`?

AssertJ API is very rich, with a lot of things possible in the call chain between `assertThat(s)` and `isNotNull()`. I can't confidently say that any such call chain should make `s` `@NonNull`.

`as()` and `describedAs()` is the thing I noticed when enabling NullAway in our codebase. Supporting just this case would address our needs.
